agent/tool-scripts/pprof

```diff
@@ -101,88 +102,75 @@ done
+            if (( i++%${#masters[@]}==0 )); then
+		for pid in ${master_pids[*]}; do
+                    wait ${pid}
+                done
+            fi
+        done
+    done
 }
 
-collect_data() {
+get_hosts() {
+    declare -A node_pids
     while true; do
-        if [ $(systemctl list-unit-files | grep atomic-openshift-master.service | awk {'print $2'}) == "masked" ]; then
-            $tool_bin tool $tool -text -seconds=$interval -output $PPROF_TMPDIR/$(date +%Y-%m-%d:%H:%M:%S)_controller.txt http://localhost:6060/debug/pprof/profile 
-            $tool_bin tool $tool -text -seconds=$interval -output $PPROF_TMPDIR/$(date +%Y-%m-%d:%H:%M:%S)_controller_heap.txt http://localhost:6060/debug/pprof/heap 
-            $tool_bin tool $tool -text -seconds=$interval -output $PPROF_TMPDIR/$(date +%Y-%m-%d:%H:%M:%S)_api.txt http://localhost:6061/debug/pprof/profile
-            $tool_bin tool $tool -text -seconds=$interval -output $PPROF_TMPDIR/$(date +%Y-%m-%d:%H:%M:%S)_api_heap.txt http://localhost:6061/debug/pprof/heap 
-        else
-            $tool_bin tool $tool -text -seconds=$interval -output $PPROF_TMPDIR/$(date +%Y-%m-%d:%H:%M:%S).txt http://localhost:6060/debug/pprof/profile 
-            $tool_bin tool $tool -text -seconds=$interval -output $PPROF_TMPDIR/$(date +%Y-%m-%d:%H:%M:%S)_heap.txt http://localhost:6060/debug/pprof/heap 
-        fi
-    done 
-} 
+        j=1
+        for host in "${hosts[@]}"; do
+            oc get --raw /debug/pprof/profile?seconds=${interval} --server https://${host}:10250 > ${host}_profile && ${tool_bin} tool ${tool} -text -seconds=${interval} -output $PPROF_TMPDIR/${host}/$(date +%Y-%m-%d:%H:%M:%S)_profile.txt ${host}_profile &
+            node_pids[${host}]=$!
+            oc get --raw /debug/pprof/heap --server https://${host}:10250 > ${host}_heap && ${tool_bin} tool ${tool} -text -seconds=${interval} -output $PPROF_TMPDIR/${host}/$(date +%Y-%m-%d:%H:%M:%S)_heap.txt ${host}_heap &
+            if (( j++%${#hosts[@]}==0 )); then
+		for pid in ${node_pids[*]}; do
+                    wait ${pid}
+                done
+            fi
+        done
+    done
+}
+
+## check if the inventory variable is set correctly
+if [[ ! -s "$inventory" ]] || [[ ! -f "$inventory" ]]; then
+	echo "Please check if the inventory file exists, is not empty and the inventory variable is not set to a directory"
+	exit 1
+fi
+
 case "$mode" in 
     install)
         check_install_rpm $tool_package 
-        ose_pprof
     ;; 
     start)
-        mkdir -p $tool_output_dir
+    	for master in "${masters[@]}"; do
+    	    mkdir -p $tool_output_dir/${master}
+    	done
+    	for host in "${hosts[@]}"; do
+    	    mkdir -p $tool_output_dir/${host}
+    	done
         collect_data
     ;; 
     stop)
         # killing pprof processes at end on of itteration 
         for pprof_process in $(ps aux | grep pbench-tool-$tool | grep -v grep | awk '{print $2}'); do kill -9 $pprof_process; done
+        rm *_profile && rm *_heap
```

Comment:
Is there a way we can give the `rm` command an explicit path to the location of where those `*_profile` and `*_heap` files live instead of assuming they are in the current directory?

And don't we want to always remove both sets of files?

Reply:
We could give the `rm` an explicit path but we are currently piping the output of the `oc get raw` to the same (current execution) directory as well. Should we put them all in `/tmp`?

Reply:
Temporary files are now in `/tmp/`